lib/services/elasticsearch.js

```diff
@@ -1109,6 +1137,35 @@ class ElasticSearch extends Service {
     return this._mexecute(esRequest, extracted.documents, extracted.rejected);
   }
 
+  /**
+   * Create or replace multiple documents at once.
+   * Does not adds additional metadata
+   *
+   * @param  {Request} request - Kuzzle API request
+   * @return {Promise}
+   */
+  mwriteDocuments(request) {
+    const
+      esRequest = initESRequest(request, ['consistency', 'refresh', 'timeout', 'fields']),
+      extracted = extractMDocuments(request, {});
+
+    esRequest.body = [];
+
+    let i; // NOSONAR
+    for (i = 0; i < extracted.documents.length; i++) {
```

Comment:
Just because I can't find another thing to say :
Feel weird to read `i++`, maybe you should always use the pre-increment syntax `++i` even if in javascript there is apparently no difference compare to C/C++/Java, I think this is good habit.

Reply:
Now I'm curious: care to explain why tihs should be a good practice?
I have so many examples right now where, unless you're code golfing, the only good habit with that operator is to make sure that its use is isolated...

Reply:
For me the good practice is to always use `+=` instead of `++` because it standardizes the code (all the incrementation are the same).
In the `++` case, it can make potential errors when you don't know the difference (but I think everyone knows the difference here) between pre and post incrementation and you could make code like this:
```javascript
const y = x++;
```

Reply:
There is almost no difference in Node.js and when I'm using iterator in C++ I didn't forget ;)
<details><summary>Benchmarks</summary>
Node.js 6
```
i++ x 4,154 ops/sec ±0.39% (89 runs sampled)
++i x 4,934 ops/sec ±0.29% (96 runs sampled)
Fastest is ++i
```
Node.js 8
```
i++ x 11,567 ops/sec ±0.14% (97 runs sampled)
++i x 11,493 ops/sec ±0.33% (96 runs sampled)
Fastest is i++
```
</details>

Reply:
@scottinet I think its a good practice because if you write it always the same way you won't forget it when you are with low languages where it has its importance. That is just an opinion :)

Reply:
![What is this, I don't even](https://i2.kym-cdn.com/photos/images/facebook/000/133/666/beeker-what-is-this-i-dont-even.gif)

First, it's very unlikely that there is a difference in performances between the pre- and post-increment operators, whatever the language. And even if there is a slight difference (again, very unlikely), it would be premature optimization to bother about it.

Second, there is nothing to remember but this one rule: do not ever mix pre-/post- increment operators with anything else in the same line to prevent unpredictability.
For instance: `foo(i, ++i)` => what are the values passed to "foo"? 

Now that's out of the way: writing the same thing always the same way doesn't seem a bad idea to me, so I can hear the consistency argument, but it's a pretty minor one to me regarding that operator, since, well, it should always be alone.

Reply:
I was talking about loops only for the "pre-increment reflex", maybe I should have precised to avoid you writing an entire paragraph :p

Reply:
Guys please leave my PR alone :see_no_evil: